migrations/zero_date_fix.go

```diff
@@ -0,0 +1,269 @@
+		if err = updateBuildCreateTime(session.DB(database), versionID, *newTime); err != nil {
+			return errors.WithStack(err)
+		}
+		if err = updateTaskCreateTime(session.DB(database), versionID, *newTime); err != nil {
+			return errors.WithStack(err)
+		}
+		if err = updatePatchCreateTime(session.DB(database), versionID, *newTime); err != nil {
+			return errors.WithStack(err)
+		}
+
+		return session.DB(database).C(versionCollection).UpdateId(versionID,
+			bson.M{
+				"$set": bson.M{
+					createTimeKey: *newTime,
+				},
+			})
+	}
+}
+
+func githubFetchRealCreateTime(token, owner, repo, revision string) (*time.Time, error) {
+	ctx := context.Background()
+	httpClient, err := util.GetHttpClientForOauth2(token)
+	if err != nil {
+		return nil, err
+	}
+	defer util.PutHttpClientForOauth2(httpClient)
+
+	client := github.NewClient(httpClient)
+
+	commit, _, err := client.Git.GetCommit(ctx, owner, repo, revision)
+	if err != nil {
+		return nil, errors.Wrap(err, "can't fetch commit info from github")
+	}
+	if commit == nil {
+		return nil, errors.Errorf("Couldn't fetch commit on %s/%s for hash: %s", owner, repo, revision)
+	}
+
+	if commit.Committer == nil || commit.Committer.Date == nil {
+		return nil, errors.Errorf("Empty data returned while fetching commit on %s/%s for hash: %s", owner, repo, revision)
+	}
+
+	return commit.Committer.Date, nil
+}
+
+func updateBuildCreateTime(dbs db.Database, versionID string, newTime time.Time) error {
+	const (
+		buildCollection = "builds"
+
+		idKey         = "_id"
+		versionKey    = "version"
+		createTimeKey = "create_time"
+	)
+
+	change, err := dbs.C(buildCollection).UpdateAll(bson.M{
+		versionKey: versionID,
+	}, bson.M{
+		"$set": bson.M{
+			createTimeKey: newTime,
+		},
+	})
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	msg := message.Fields{
+		"message":  "updated builds",
+		"version":  versionID,
+		"new_time": newTime.String(),
+	}
+
+	if change != nil {
+		msg["updated"] = change.Updated
+	}
+	grip.Info(msg)
+
+	return nil
+}
+
+func updatePatchCreateTime(dbs db.Database, versionID string, newTime time.Time) error {
+	const (
+		tasksCollection = "tasks"
+
+		idKey         = "_id"
+		versionKey    = "version"
+		createTimeKey = "create_time"
+	)
+
+	change, err := dbs.C(tasksCollection).UpdateAll(bson.M{
+		versionKey: versionID,
+	}, bson.M{
+		"$set": bson.M{
+			createTimeKey: newTime,
+		},
+	})
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	msg := message.Fields{
+		"message":  "updated tasks (most recent execution only)",
+		"version":  versionID,
+		"new_time": newTime.String(),
+	}
+	if change != nil {
+		msg["updated"] = change.Updated
+	}
+	grip.Info(msg)
+
+	return nil
+}
+
+func updateTaskCreateTime(dbs db.Database, versionID string, newTime time.Time) error {
+	const (
+		patchesCollection = "patches"
+
+		idKey         = "_id"
+		versionKey    = "version"
+		createTimeKey = "create_time"
+	)
+
+	change, err := dbs.C(patchesCollection).UpdateAll(bson.M{
```

Comment:
I can/would use anser/db.Document{} (which like bson.M is an alias for map[string]interface{}) rather than bson.M. 

I realize that I've missed a few of these in previous reviews, and we can push without this, but it'll make it slightly easier to move to the new driver for migrations (which I expect we'll do before moving the rest of the application.)